Assets/Project/Scripts/UnityChanControlScript.cs

```diff
@@ -54,178 +63,114 @@ private void Start()
-
-            return;
         }
 
-        void ResetCollider()
+        // return keycode if mapped in our dictionary
+        private KeyCode TryGetKeyCode(string animatorParam)
         {
-            col.height = orgColHight;
-            col.center = orgVectColCenter;
+            return animationStates.ContainsKey(animatorParam) ? animationStates[animatorParam].Key : default;
         }
 
-        #region Actions depending on current state
-
-        void WhenRunning()
+        // if param exists and is enabled then call the function that it points to
+        private void PerformActionIfEnabled(string requestedAction)
         {
-            if (useCurves)
-            {
-                ResetCollider();
+            if (animationStates.ContainsKey(requestedAction) && animationStates[requestedAction].IsEnabled)
+            {               
+                animationStates[requestedAction].FuncToCall(); // invoke action in dictionary that corresponds to it
             }
+            return;
         }
 
-        void WhenIdle()
+        // ====== Set of functions that can be called if enabled =========
+        protected void JumpControls()
         {
-            if (useCurves)
+            if (!anim.IsInTransition(0))
             {
-                ResetCollider();
-            }
+                rb.useGravity = true;
 
-            if (Input.GetButtonDown("Jump"))
-            {
-                anim.SetBool("Rest", true);
-            }
+                rb.AddForce(Vector3.up * jumpPower, ForceMode.Impulse);
+                anim.SetTrigger(jumpParam);
+            }            
+        }           
+        protected void SlidingControls()
+        {            
+            anim.SetTrigger(slideParam);
         }
-
-        void WhenJumping()
-        {
-            if (!anim.IsInTransition(0))
-            {
-                if (useCurves)
-                {
-                    float jumpHeight = anim.GetFloat("JumpHeight");
-                    float gravityControl = anim.GetFloat("GravityControl");
-                    if (gravityControl > 0)
-                        rb.useGravity = false;
-
-                    Ray ray = new Ray(transform.position + Vector3.up, -Vector3.up);
-                    RaycastHit hitInfo = new RaycastHit();
-
-                    if (Physics.Raycast(ray, out hitInfo))
-                    {
-                        if (hitInfo.distance > useCurvesHeight)
-                        {
-                            col.height = orgColHight - jumpHeight;
-                            float adjCenterY = orgVectColCenter.y + jumpHeight;
-                            col.center = new Vector3(0, adjCenterY, 0);
-                        }
-                        else
-                        {
-                            ResetCollider();
-                        }
-                    }
-                }
-
-                anim.SetBool("Jump", false);
-            }
+        protected void AttackControls()
```

Comment:
These methods can be made private

Reply:
oops this was when I was trying to move them to another file, defo private